### ISSUE TITLE
ArrayIndexOutOfBoundsException at System.arraycopy in Blake2spDigest update method for message length 1007 bytes

### ISSUE THREAD

**Issue by jonelo** [bcgit/bc-java#1363]
When hashing a message length of 1007 bytes with Blake2spDigest I got the following exception:
```
java.lang.ArrayIndexOutOfBoundsException: arraycopy: last destination index 495 out of bounds for byte[256]
```
I figured out that in that case the arraycopy call failed at line 102 at
```
            System.arraycopy(message, offset, buffer, left, len);
```
because of the values offset=512, buffer.length=256, left=0, and len=495

**Comment by jonelo** [comment]
a possible fix would be to increase the buffer size to 512 bytes as in commit https://github.com/jonelo/bc-java/commit/cdaf22c87886aeac38b8afb6410ebefb3cd6443d

**Comment by dghgit** [comment]
Fixed in 1.73
